1_The_Physical/Level_2_Stateful_Guess_Point/Shared/Student/src/domain/student/value-objects/FirstName.ts

```diff
@@ -0,0 +1,31 @@
+import { Result } from "../../../shared/Result";
+import { ValueObject } from "../../core";
+import { FirstNameValidationError, FirstNameValidator } from "../validators";
+
+interface FirstNameProps {
+  value: string;
+}
+
+export class FirstName extends ValueObject<FirstNameProps> {
+  private constructor(props: FirstNameProps) {
+    super(props);
+  }
+
+  public static create(
+    value: string
+  ): Result<FirstName, FirstNameValidationError> {
+    const errors = new FirstNameValidator().validate(value);
+
+    if (
+      Object.values(errors).some((props) => Object.values(props).some(Boolean))
```

Comment:
This is repeated multiple times and is hard to read, it's a good candidate to move to a utility function and re-use.